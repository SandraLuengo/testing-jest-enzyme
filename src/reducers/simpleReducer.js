const initialState = {
  contact: '',
};

export default (state = initialState, action) => {
  switch (action.type) {
    case 'SIMPLE_ACTION':
      return {
        contact: action.newValue,
      };
    default:
      return state;
  }
};
