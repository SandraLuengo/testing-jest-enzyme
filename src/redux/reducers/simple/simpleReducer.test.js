import simpleReducer from './simpleReducer';

describe('SIMPLE_ACTION', () => {
  it('initialState is correct', () => {
    const action = { type: 'TESTING_SIMPLE_ACTION' };
    const initialState = { contact: 1 };
    // expect(simpleReducer(undefined, action)).toEqual(initialState);
    expect(simpleReducer(undefined, action)).toMatchSnapshot();
  });
  it('returns the correct state', () => {
    const action = { type: 'SIMPLE_ACTION', payload: 'pedro' };
    const expectedState = { contact: 'pedro' };
    // expect(simpleReducer(undefined, action)).toEqual(expectedState);
    expect(simpleReducer(undefined, action)).toMatchSnapshot();
  });
});

