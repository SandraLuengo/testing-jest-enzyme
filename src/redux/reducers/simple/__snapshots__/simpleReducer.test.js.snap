// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIMPLE_ACTION initialState is correct 1`] = `
Object {
  "contact": 1,
}
`;

exports[`SIMPLE_ACTION returns the correct state 1`] = `
Object {
  "contact": "pedro",
}
`;
